core/org.wso2.carbon.bootstrap/src/main/java/org/wso2/carbon/bootstrap/Bootstrap.java

```diff
@@ -68,6 +74,37 @@ protected final void loadClass(String args[]) {
 
     }
 
+    private void addSystemProperties(){
+        Properties properties = new Properties();
+        String filePath = ROOT + File.separator + "repository" + File.separator
+                + "conf" + File.separator + CARBON_PROPERTIES;
+        File file = new File(filePath);
+
+        if (file.exists()) {
+            InputStream in = null;
+            try {
+                in = new FileInputStream(file);
+                properties.load(in);
+            } catch (IOException e) {
+                e.printStackTrace();
+                System.exit(1);
+            } finally {
+                if (in != null) {
+                    try {
+                        in.close();
+                    } catch (IOException ignored) {
+                        // Exception is ignored as there is no need to break the execution here
+                    }
+                }
+            }
```

Comment:
Better if we can use try-with-resource to read properties file. we can reduce no of code lines. refer[1]. WDYT?
1. http://tutorials.jenkov.com/java-exception-handling/try-with-resources.html